styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -628,9 +629,11 @@ private void emitPodEvents(Pod pod, RunState runState) {
         // TODO: spoofing counter values like this can give unexpected results, e.g. if we emit two events here the
         // first one might be discarded and the second one accepted.
         stateManager.receive(event, runState.counter() + i);
-      } catch (IsClosedException isClosedException) {
-        LOG.warn("Could not receive kubernetes event", isClosedException);
-        throw new RuntimeException(isClosedException);
+      } catch (StateTransitionConflictException e) {
+        LOG.debug("State transition conflict on kubernetes pod event: {}", event, e);
+        return;
+      } catch (IsClosedException ignore) {
```

Comment:
We can use `receiveIgnoreClosed` if we don't care this exception.

Reply:
I wanted to exit the loop if we got IsClosedException, but I guess that's not really necessary.

Reply:
Changing this to use `receiveIgnoreClosed` requires changing test cases as well, so opting to keep this as-is.